frontend/src/app/annotate/annotate.component.ts

```diff
@@ -15,10 +21,51 @@ import { faTree } from "@fortawesome/free-solid-svg-icons";
         AnnotationInputComponent,
         SearchComponent,
         FontAwesomeModule,
+        CommonModule,
     ],
     templateUrl: "./annotate.component.html",
     styleUrl: "./annotate.component.scss",
 })
-export class AnnotateComponent {
-    public faTree = faTree;
+export class AnnotateComponent implements OnInit {
+    private router = inject(Router);
+    private route = inject(ActivatedRoute);
+    private problemService = inject(ProblemService);
+    private destroyRef = inject(DestroyRef);
+
+    public faPlus = faPlus;
+    public faBinoculars = faBinoculars;
+
+    public appMode$ = this.problemService.appMode$;
+    public firstProblemId$ = this.problemService.firstProblemId$;
+
+    public isUserProblem$ = this.problemService.problem$.pipe(
+        map(problem => problem?.dataset === Dataset.USER)
+    );
+
+    ngOnInit(): void {
+        const editParam$ = this.route.url.pipe(
+            map(segments => segments.some(segment => segment.path === "edit")),
```

Comment:
I don't actually recommend that you do this because the project is not yet using Underscore, but I always like to point out nice ways to enable shorter notation like this:

```suggestion
            map(segments => _.some(segments, {path: "edit"})),
```

(You could even do `map(_.partial(_.some, _, {path: 'edit'}))`, though that is admittedly less recognizable for most people.)

Reply:
Interesting. Would the type checker know about `path` has to be a property of an element of `segments`? Would it complain if I accidentally wrote `ptah` (or some other Egyptian god)?

Reply:
The people who wrote `@types/underscore` (and `/lodash`) went to great lengths to enable TS to infer such things. In this case, I think it might be able to catch the typo. That being said, JS is at its core a dynamically typed language and some of its types are just untraceable without running the program. Underscore has not restricted itself to the subset of JS that TS can accurately type. For example, I know that `_.flatten` has a type that TS cannot express because of the nested infinities involved.